extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -310,6 +312,52 @@ export class AmpImageSlider extends AMP.BaseElement {
     this.bar_.appendChild(this.hint_);
   }
 
+  /**
+   * Check if aria attributes are correctly set
+   * If not, apply default and warn user in console
+   * @private
+   */
+  checkARIA_() {
+    const leftAmpImage = dev().assertElement(this.leftAmpImage_);
+    const rightAmpImage = dev().assertElement(this.rightAmpImage_);
+    leftAmpImage.signals().whenSignal(CommonSignals.LOAD_END).then(() => {
```

Comment:
Do we want to be waiting for `LOAD_END` in build callback? Ideally, buildCallback should finish very quickly and not be waiting for anything to finish layout. Let's move this to layoutCallback. 

There's something slightly annoying about the `LOAD_END` signal. It's actually just an event trigger, so for example if wait for `LOAD_END` on an element that has already loaded, it won't actually fire again. What are you are really hoping to wait for here is for the `amp-image` to finish layout. So if the element contains `i-amphtml-layout` as an attribute or a classname, you won't need to wait for `LOAD_END`. This is applicable for server side rendering as well as for potential bugs where you have an owner, and it calls layout / unlayout callback on your behalf.

Reply:
Right. I don't disagree with the use of `LOAD_END`. However, it's not always the case that we will necessarily signal `LOAD_END`. There have been bugs where layoutCallback was just not called. 
https://github.com/ampproject/amphtml/blob/22467233e8e80336d164984ed2afa60ddb6e4d24/src/custom-element.js#L1070-L1073. 

But in the particular cases of it that have occurred, it's due to element cloning or styling issues that causes the `amp-img` to be 0x0. Neither is applicable here, unless the user decides to manipulate the visibility/dimensions of the `<amp-img>`s with CSS, so we should be fine.